client/verta/verta/monitoring/summaries/aggregation.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+
+from verta._internal_utils import time_utils
+from verta._protos.public.monitoring.Summary_pb2 import AggregationQuerySummary
+from verta.external import six
+
+
+class Aggregation(object):
+    """A query object specifying how summary samples should be aggregated.
+
+    Attributes
+    ----------
+    granularity : :class:`datetime.timedelta`
+        The granularity of aggregation.
+    operation : str
+        The aggregation operation to perform.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        from datetime import timedelta
+        from verta.monitoring.summaries.aggregation import Aggregation
+
+        weekly_agg = Aggregation(timedelta(days=7), "sum")
+        hourly_agg = Aggregation("1h", "sum")
+        minute_agg = Aggregation(1000 * 60, "sum")
+
+    """
+
+    _OPERATIONS = {
+        k.lower(): v for k, v in AggregationQuerySummary.AggregationOperation.items()
+    }
+
+    def __init__(self, granularity, operation):
+        self.granularity = granularity
+        self.operation = operation
+
+    def __repr__(self):
+        return "Aggregation('{}',{})".format(self.granularity, self.operation)
+
+    @property
+    def granularity(self):
+        """The granularity of aggregation.
+
+        :getter: Returns the specified time granularity of aggregation
+        :setter: Sets the time granularity from a timedelta, string, or int of epoch milliseconds.
```

Comment:
I have absolutely no clue how Sphinx's autosummary handles docstrings inside properties 😬 But it's probably fine (worst case, the class docstring is descriptive enough for this to be used from looking at the API docs).